README.md

```diff
@@ -31,38 +33,64 @@ const client = new SgidClient({
 })
 ```
 
+### Generate PKCE pair
+
+`generatePkcePair([length])`
+
+```typescript
+const { codeChallenge, codeVerifier } = generatePckePair()
+```
+
 ### Get Authorization URL
 
-`client.authorizationUrl(state, scope, [nonce], [redirectUri])`
+`async client.authorizationUrl(parameters)`
+
+- parameters: `<Object>`
+  - codeChallenge: `<string>`
+  - state: `<string>` (Optional)
+  - scope: `<string | string[]>` (Optional) **Default**: `myinfo.name openid`
+  - nonce: `<string> | <null>` (Optional) **Default**: Randomly generates a nonce if unspecified
+  - redirectUri: `<string>` (Optional) **Default**: Utilizes the `redirectUri` provided in the constructor
 
 ```typescript
-const { url } = client.authorizationUrl(
-  'state',
-  ['openid', 'myinfo.nric_number'], // or space-concatenated string
-  null, // defaults to randomly generated nonce if unspecified
-  'http://localhost:3000/other_callback', // overrides redirect uri
-)
+const { url } = client.authorizationUrl({
+  codeChallenge: 'zaqUHoBV3rnhBF2g0Gkz1qkpEZXHqi2OrPK1DqRi-Lk', // generated from the previous step
+  state: 'state', // optional as PKCE protects against CSRF
```

Comment:
we can also reduce the number of unfamiliar terms used here. if user doesn't need to worry about CSRF, we don't need to mention it. instead, we can say something like "for you to track application state which will be passed back to you after login"

Reply:
mm yeah that makes sense